### DISCUSSION TITLE
Can I get the name of the active Stream Deck profile?

### DISCUSSION THREAD

**Discussion by MillerMark** [elgatosf/elgato#67] [Help] [answered]
I need to get the name of the active profile page. Is there any way to do this?

Use case: My install includes nine different profiles to extend Visual Studio development. I have profiles that automatically switch in response to certain events in Visual Studio. For example, when you build, the "VS Build" profile is activated. When you debug, the "VS Debug" profile is active. And when you give focus to Visual Studio, profiles may switch based on context (what's happening in the Visual Studio instance receiving focus). Some of these profiles include buttons that are folders, and I noticed that if I'm in the "VS Debug" profile but inside a folder, and if I then I activate Visual Studio, that my Stream Deck pops me out of that folder (and any parenting folders) so I arrive at a top-level page, seemingly in response to my attempt to activate the "VS Debug" profile (which is already active). Some might argue that this is a bug (if the profile I'm switching to is already active, why would it navigate the user out of folder they were in?). Regardless, if I could determine that the profile I want to switch to was **already active**, I could avoid calling SwitchToProfile, which could provide a smoother, less disruptive experience for the user.

**Comment by ChekTek** [comment, marked as answer]
Hi @MillerMark, 

While it is not possible at this time, this API change has been request in the past, and we would like to add to Stream Deck. Unfortunately, it could take some time before this feature is delivered as our Stream Deck team is working hard to delivery more features. I will add your request to the ticket to encourage prioritizing it for the future.
